ci/start_init_test.sh

```diff
@@ -74,16 +80,81 @@ if [ "$x"!="" ]; then
         fi
 
         # TEST#3: Run the libiscsi compliance suite on Jiva Vol
-        sudo mkdir /mnt/logs 
+        sudo mkdir /mnt/logs
         sudo docker run -v /mnt/logs:/mnt/logs --net host openebs/tests-libiscsi /bin/bash -c "./testiscsi.sh --ctrl-svc-ip 172.18.0.2"
         tp=$(grep "PASSED" $(find /mnt/logs -name SUMMARY.log) | wc -l)
         tf=$(grep "FAILED" $(find /mnt/logs -name SUMMARY.log) | wc -l)
         if [ $tp -ge 146 ] && [ $tf -le 29 ]; then
-            echo "iSCSI Compliance test: PASSED"
-        else 
+           echo "iSCSI Compliance test: PASSED"
+        else
             echo "iSCSI Compliance test: FAILED"; exit 1
         fi
-else 
+else
         echo "Unable to detect iSCSI device, login failed"; exit 1
-fi 
+fi
+
+sudo umount /mnt/store
+sudo iscsiadm -m node -u
+sudo iscsiadm -m node -o delete
+
+# Test clone feature
+sudo docker run -d --net stg-net --ip 172.18.0.5 -P --expose 3260 --expose 9501 --expose 9502-9504 $JI launch controller --frontend gotgt --frontendIP 172.18.0.5 store1
+sudo docker run -d -it --net stg-net --ip 172.18.0.6 -P --expose 9502-9504 -v /tmp/vol3:/vol3 $JI launch replica --type clone --snapName snap1 --cloneIP 172.18.0.2 --frontendIP 172.18.0.5 --listen 172.18.0.6:9502 --size 2g /vol3
+
+clonestatus=`curl http://172.18.0.6:9502/v1/replicas/1 | jq '.clonestatus' | tr -d '"'`
+
+i=0
+while [ -z $clonestatus ]; do
+        sleep 3
+	clonestatus=`curl http://172.18.0.6:9502/v1/replicas/1 | jq '.clonestatus' | tr -d '"'`
+        i=`expr $i + 1`
+        if [ $i -eq 20 ]; then
+		echo "Clone process took longer than usual"
+                exit 1;
+        else
+		echo "Waiting for clone process to complete"
+                continue
+        fi
+done
+
+# Display running containers
+sudo docker ps
+
+# Display volume info
+curl http://172.18.0.5:9501/v1/volumes
+
+# Discover Jiva iSCSI target and Login
+sudo iscsiadm -m discovery -t st -p 172.18.0.5:3260
+sudo iscsiadm -m node -l
+
+# Wait for iSCSI device node (scsi device) to be created
+sleep 1
+sudo fdisk -l
+x=$(iscsiadm -m session -P 3 |grep -i "Attached scsi disk" | awk '{print $4}')
+echo $x
+
+i=0
+while [ -z $x ]; do
+        sleep 4
+        x=$(iscsiadm -m session -P 3 |grep -i "Attached scsi disk" | awk '{print $4}')
+        i=`expr $i+1`
+        if [ $i -eq 5 ]; then
+                break
+        else
+                continue
+        fi
+done
+
+if [ "$x"!="" ]; then
+# Mount FS onto local mountpoint
+	sudo mount /dev/$x /mnt/store
 
+	# TEST#1: Perform simple data-integrity check on Jiva Vol
+	hash3=$(sudo md5sum /mnt/store/file1 | awk '{print $1}')
+	if [ $hash1 == $hash3 ]; then echo "DI Test: PASSED"
+	else
+		echo "DI Test: FAILED"; exit 1
+	fi
+else
+	echo "Unable to detect iSCSI device, login failed"; exit 1
+fi
```

Comment:
How is this test run automatically?

Reply:
I didn't get the question. If you are asking from where this shell script is triggered, its from the Makefile, after the image is built.